File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/gui/GuiChat.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.gui;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapField;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;

import java.lang.reflect.Field;

@WrapperClass(mcpName = "net.minecraft.client.gui.GuiChat", targetMap = MapEnum.VANILLA189)
public class GuiChat extends GuiScreen {
    @WrapClass(mcpName = "net.minecraft.client.gui.GuiChat", targetMap = MapEnum.VANILLA189)
    public static Class GuiChatClass;
    @WrapField(mcpName = "inputField", targetMap = MapEnum.VANILLA189)
    public static Field inputField;
    @WrapField(mcpName = "sentHistoryCursor", targetMap = MapEnum.VANILLA189)
    public static Field sentHistoryCursor;

    public GuiChat(Object obj) {
        super(obj);
    }

    public static boolean isGuiChat(GuiScreen guiScreen) {
        return GuiChatClass.isInstance(guiScreen.getWrapObject());
    }

    public GuiTextField getInputField() {
        return new GuiTextField(getField(inputField));
    }

    public void setInputField(GuiTextField textField) {
        setField(inputField, textField.getWrapObject());
    }

    public void setSentHistoryCursor(int i) {
        setField(sentHistoryCursor, i);
    }
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/gui/GuiChat.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.gui

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapClass
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapField
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass
import java.lang.reflect.Field

@WrapperClass(mcpName = "net.minecraft.client.gui.GuiChat", targetMap = MapEnum.VANILLA189)
class GuiChat(obj: Any) : GuiScreen(obj) {
    var inputField: GuiTextField
        get() = GuiTextField(getField(Companion.inputField)!!)
        set(textField) {
            setField(Companion.inputField, textField.wrapObject)
        }

    fun setSentHistoryCursor(i: Int) {
        setField(sentHistoryCursor, i)
    }

    companion object {
        @WrapClass(mcpName = "net.minecraft.client.gui.GuiChat", targetMap = MapEnum.VANILLA189)
        lateinit var GuiChatClass: Class<*>

        @WrapField(mcpName = "inputField", targetMap = MapEnum.VANILLA189)
        lateinit var inputField: Field

        @WrapField(mcpName = "sentHistoryCursor", targetMap = MapEnum.VANILLA189)
        lateinit var sentHistoryCursor: Field

        @JvmStatic
        fun isGuiChat(guiScreen: GuiScreen): Boolean {
            return GuiChatClass.isInstance(guiScreen.wrapObject)
        }
    }
}